### DISCUSSION TITLE
[Open Discussion] Best options for forming a set leveling and raid team?

### DISCUSSION THREAD

**Discussion by Dreathean** [mod-playerbots/mod-playerbots#186] [Ideas]
**Is your feature request related to a problem? Please describe.**

No issue, just wanting to discuss with fellow players to determine the best currently available options to form a set team for leveling and eventually raiding. My goals are a team/guild of 25 (including myself and a friend) with specifically chosen class, names, race, etc that will level as we level and can pull some as needed for dungeons, and then ultimately use the same team for raiding. Persistence in these playerbots is key.

Currently there seem to be three options: 
1) Create alts and allow playerbots to control them via .playerbots bot add (name). This does allow you to automatically create the name, race, and gender that you want. However, unless I'm missing something there are serious drawbacks: limited to 9 alts and they do not get all of the "randombot" abilities like free food, talent choices, automatic gearing while leveling, etc. Currently does not seem feasible for my plans.

2) Use the playerbots bot addclass (class) option. This allows you to generate a full randombot of the class you want, however main drawbacks are you can't choose name, race, or gender. However I suppose you should be able to change all 3 to your liking in the SQL database, correct? Then you can add them all to your friendlist and use the friends command to make them all come online when you wish. Not 100% sure how that command interacts with real friends however, hopefully that doesn't try to make real characters on your friends list activated via playerbots.

3) Basically the same as 2 except use various bots automatically generated in your set amount of randombots and adjust as needed. To be honest not sure of the pros and cons of doing one or the other.

At this time I'm planning on using option 2 and hoping it won't be difficult to adjust the names, race, and gender as desired in the database. Am I missing anything? Appreciate any and all ideas and insight!

**Comment by Dreathean** [comment]
Slight update: in testing option 2, the randombots via addclass command only seem to want to follow me even after uninviting them from party. If I teleport somewhere else, they just stand where they last were. Tried resetAI but that didn't seem to do anything. Not sure what to do here? Would like them to resume regular randombot activities after uninviting them from party.

**Comment by liyunfan1223** [comment]
My opinion is option 1 should be the best way :D, or combine both options 1 and 2.

You can invite characters from other accounts to the same guild and set `AllowGuildBots=1` to control more than 9 bots.

After inviting them, using the `maintenance` command can automatically make them prepare almost everything before the raid (without initialized equipment), like learning available spells, skills, talents, and glyphs, initializing food and reagents, enchanting and repairing, etc. The `equip upgrade` command makes them choose better items in their bag to equip (this function is a work in progress, they may be unable to choose the best item so far).

And option 2 can be a supplement when we don't have enough characters.

**Reply by Dreathean** [reply]
Hey @liyunfan1223 ! I was trying this out, and even with AllowGuildBots = 1, I was unable to add another account's characters as bots.
![WoWScrnShot_041924_214941](https://github.com/liyunfan1223/mod-playerbots/assets/132776330/e8f4380a-28d1-4ee2-adc5-84a6aee944be)
![WoWScrnShot_041924_214953](https://github.com/liyunfan1223/mod-playerbots/assets/132776330/c1ff5316-631a-4306-bdad-bdfc15050223)

Tried both while one was offline and another online. Hopefully this would be a simple fix?

**Reply by liyunfan1223** [reply]
Also try setting `AiPlayerbot.AllowPlayerBots=1`, which added in https://github.com/liyunfan1223/mod-playerbots/pull/171.

**Reply by liyunfan1223** [reply]
It should work. It looks like these configurations have become repetitive and unintuitive. I'm thinking of making them simpler.

**Reply by Dreathean** [reply]
Ah I had forgotten to add the AiPlayerbot.AllowPlayerBots=1 setting to my .conf when that was added. Okay so that mostly works, though I noticed if a character name is the same as the account name (separate account), then the .playerbot bot add "name" will try to do something with the account instead of the character, not sure if that is fixable or not?

**Reply by Dreathean** [reply]
I also noticed that when I logout on the character that added all the alt-characters as bots, they are logged out and would have to be manually re-added as bots everytime I log back in. Is there any way to make that persistent? Appreciate all the help!!

Edit: worst case I imagine I can make a macro to tag the whole raid team and another to maintenance each of them

**Comment by Dreathean** [comment]
Good to know about using extra accounts and guild to control more alt-created playerbots!

I forgot about the new maintenance command - that is huge and definitely solves a lot of those issues. And presumably these alt-created playerbots are safe from any potential deletion when reseting randombots. 

However, it seems that this does not include the automatic gearing upon level-up (not factoring in the level cap equipment persistence)? Or will they be able to do that as well? If not we can work around that by just using randombots during leveling and then generate the alt-bots when we're ready to raid, but then that would also still involve gearing them up for starter gear. Which is fine if that is the case, just seeing if it is possible for these alt-bots to gear themselves up to the equipment persistence level cap. It would just be cool to watch the progress on your guildie alt-bots alongside your own leveling and then seamlessly use them to raid you with haha.

**Comment by liyunfan1223** [comment]
Yeah, an easier starter gearing makes sense. Perhaps it's better to implement it as another command to avoid overusing it.

**Comment by Dreathean** [comment]
Yup, the main thing is that it is simply not feasible to manually keep 24 other bots geared up during the leveling process (raiding is of course different and I intend to handle that manually). Would it be possible to implement a command similar to the maintenance command that allows them to use the same auto-gearing process randombots use while leveling up?

**Comment by Mr-Deadbeat** [comment]
i'm using the same approach i play with 9 other characters on my account also with the mindset of raiding with them one day
once they have a little more functionality atm i'm just waiting for them to be able to complete quests / pickup quest items / interact with quest objects had lots of difficulties with BRD / LBRS / UBRS but since i'm a GM i could just do .quest complete on them tedious but only solution buddy of mine is not so lucky and i sure as hell are not going to give him GM he tends to do shit he is not supposed to so would probably break the server if given too much power ;)

getting a bit sidetracked here lol anywhoo my plan is to raid 10 man content first once all my characters have reached level 80 i had some thoughts about trying Zul'aman / Kharazan / Gruul / Magtheridon before that but i don't think the bots will be ready for that yet unless there are Strategies in place for those raid instances one of the bosses in Zul'aman will cause issues for sure i do not remember his name atm but it has to do with standing in either of 4 corners when he starts channeling his AOE if you slip up there you are screwed ;)

later on i might try BWL since i have never seen or done that raid before so feels like i have missed some content also Black Temple comes to mind i did see some of parts of the instance back in the day when i still played on retail but that was ages ago so forgotten a lot there are other raids as well i might try but for the time being i'm happy if i can test / do the 10 man raids at least :)

should add that i will consider starting from scratch "lvl 1" once bots have more functionality to give me that feeling of accomplishment so at 60 aim for the 60 raids and at 70 move to the outlands ones and so on atm i have nothing but time unemployed ;)

**Comment by Dreathean** [comment]
For most of the early raids, I found during our NPCbots run, I would just have to go in and tune a handful of spells per raid to adjust bosses with spells that npcbots (and now playerbots) aren't equipped to handle for whatever reason. It's very few for vanilla especially. 

Hopefully a command to add the auto-gear generation to altbots can be added so that you can accessibly share the leveling journey with your future raid team!

**Reply by Mr-Deadbeat** [reply]
nah i would not use any kind of auto generating to get gear for my Alt characters maybe for random bots but not for the ones i play with :) you get the feeling of accomplishment when you have earned the gear you are wearing ;) with trickerers bots before all the changes to spell casting so they stopped working on npcbots i was working on Heroics in Northrend to gear my characters up was a bit more work since you could not run around with your Alts but you still got a sense of accomplishment if you know what i mean clearing Pit of Saron on heroic for the first time with the bots felt great lots of cursing though since trickerers bots are hard to play with since you need to micro manage them with macros

and towards the end when i gave up on them there were simply too many macros needed paladin felt totally neutered since his spells did not work on npcbots unless you jumped trough hoops and created macros so got to be a real pain in the ass

with the playerbots there is hope that one day you get to truly experience the instances almost like playing with real players :)
atm i have like macros for summoning team / logoff / Strategy / looting / no looting / loot quality once that functionality is added
atm the bots pickup every bit of scrap along the way so bags fill up in no time at all either way with the above macros compared to 20+ for managing trickerers bots that is a change for the better if you ask me ;)

also getting jumped by random bots from opposing faction makes traveling a bit more interesting when you do not know if something is going to try to hit you over the head lol and since they move about you never know when and where ;)

**Reply by Dreathean** [reply]
I mean, to each their own, which is why it would ideally be a toggleable config option like the maintenance command. Personally I'm not going to manually gear 23 alt bots all through 1-60 except when we get near 60. It will be enough satisfaction and sense of accomplishment for me and my buddy to gear ourselves up while leveling and then gear up our team for the last leg at 55-60 and then of course achieve raid gear.

**Reply by Mr-Deadbeat** [reply]
yeah i can understand that i have no intention of gearing up more than my 10 characters on alliance side and 10 on horde side
created accounts for both factions so i can spot if there are buggy quests or other issues on either side my buddy will probably have 10 at some point as well so when we can't raid together he can do 10 mans solo the rest of the bots will be invited from the random bots i have added to the server i have 80 accounts and using the pool function so they get switched every now and then tested adding random bots on my horde account to the guild and so far they have been present they come and go like normal players lol so kind of cool ;) sure hope i can get them when it's time to raid :P

**Comment by abracadaniel222** [comment]
It would be great if the altbots could behave more closer to rndbots. At the very least: auto quest accept, progress, complete when in a group with the real player, just like rndbots. **But ideally**: being also able to just do quests, level up and everything on their own like rndbots do.

**Reply by ruckerzerg** [reply]
Why is that not possible? Should be easy as the logic is already there, isn't it? Would also be cool if to send my alts do some leveling on their own while I play alone.

**Reply by Dreathean** [reply]
It’s just not yet. Currently the only real option for altbots to level on their own is to invite them into a raid, type “/r nc+ grind,+new rpg” then disband the raid, and they will seek nearby quests and mobs to grind. But it’s not to the degree of rndbots and it’s still a bit messy. You would also have to do that process every time you or the bots log in and out.
